addons/stock/models/stock_picking.py

```diff
@@ -377,6 +380,12 @@ def _compute_has_tracking(self):
         for picking in self:
             picking.has_tracking = any(m.has_tracking != 'none' for m in picking.move_lines)
 
+    @api.depends('move_lines.delay_alert_date')
+    def _compute_delay_alert_date(self):
+        for picking in self:
+            sorted_move_lines = picking.move_lines.filtered(lambda m: m.delay_alert_date).sorted(lambda m: m.delay_alert_date)
+            picking.delay_alert_date = sorted_move_lines[-1].delay_alert_date if sorted_move_lines else False
```

Comment:
At this point, it would be interesting to check the date on the picking and the picking type delivery policy (ASAP, all at once). It's not true that the picking is always late if one move is late.

Reply:
@amoyaux the date of the moves will always modify `scheduled_date` on the stock picking, so it can be interesting to show the information whatever the delivery policy

what do you think @sle-odoo ?

Reply:
we ignore the asap/all at once concept on these alerts at the moment